test/Microsoft.Azure.Devices.Edge.Test.Common/LeafDevice.cs

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Test.Common.Certs;
+    using Microsoft.Azure.Devices.Edge.Test.Common.Config;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
+
+    public class LeafDevice
+    {
+        readonly DeviceClient client;
+        readonly Device device;
+        readonly IotHub iotHub;
+        readonly string messageId;
+
+        LeafDevice(Device device, DeviceClient client, IotHub iotHub)
```

Comment:
similar to edgeModule, should we remove IotHub argment here?

Reply:
LeafDevice uses `this.iotHub` in three of its four non-static public methods. It needs `this.iotHub`.